services/user_service.go

```diff
@@ -202,3 +212,93 @@ func (u *UserService) UpdatePassword(ctx context.Context, data *models.UpdatePas
 
 	return user, nil
 }
+
+func (u *UserService) GeneratePasswordResetToken(ctx context.Context, baseURL string, data *models.ForgotPassword) error {
+	var resetToken string
```

Comment:
why is reset token defined here?

Reply:
oh I'm using a variable because I pass it into the `sendPasswordResetEmail` function as well.

Reply:
Yeah, but it's initialized on line 230 so there's not need to define it here

Reply:
okay I see your point.